cocotb/triggers.py

```diff
@@ -734,3 +734,35 @@ def _wait(self):
         for _ in range(self.num_cycles):
             yield trigger
         raise ReturnValue(self)
+
+
+@decorators.coroutine
+def with_timeout(triggers, timeout_time, timeout_unit=None):
+    """
+    Waits on triggers, throws an exception if it waits longer than the given time
+
+    Results are the first trigger that completed, or a SimTimeoutError if timeout
+    occured.
+
+    Usage:
+        yield with_timeout(coro, 100, 'ns')
+        yield with_timeout([coro, event.wait(), ...], 100, 'ns')
+
+    Args:
+        triggers : yield_expr
+            Any expression that could be right of a yield statement in cocotb
+        timeout_time : int
+            Time duration
+        timeout_unit : Optional[str]
+            Units of duration, accepts any values that cocotb.triggers.Timer does
```

Comment:
```suggestion
            Units of duration, accepts any values that :class:`Timer` does
```
will symlink this nicely.